src/Cardano/Mnemonic.hs

```diff
@@ -263,3 +280,95 @@ mnemonicToText =
     . unListN
     . mnemonicSentenceToListN
     . mnemonicToSentence
+
+data SomeMnemonic where
+    SomeMnemonic :: forall mw. KnownNat mw => Mnemonic mw -> SomeMnemonic
+
+deriving instance Show SomeMnemonic
+instance Eq SomeMnemonic where
+    (SomeMnemonic mwa) == (SomeMnemonic mwb) =
+        case typeOf mwa `testEquality` typeOf mwb of
+            Nothing -> False
+            Just Refl -> mwa == mwb
+
+-- | Create a passphrase from a mnemonic sentence. This class enables caller to
+-- parse text list of variable length into mnemonic sentences.
+--
+-- >>> fromMnemonic @'[12,15,18,21] @"generation" ["toilet", "curse", ... ]
+-- Right (Passphrase <ScrubbedBytes>)
```

Comment:
That comment is actually outdated. `FromMnemonic` used to return a `Passphrase` but it now returns `SomeMnemonic`.

Reply:
corrected